scripts/validate_topbraid_uris.py

```diff
@@ -0,0 +1,157 @@
+
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+        PREFIX cdd:<http://data.mskcc.org/ontologies/clinical_data_dictionary/>
+        PREFIX skos:<http://www.w3.org/2004/02/skos/core#>
+        SELECT ?column_header ?display_name ?attribute_type ?datatype ?description ?priority
+        WHERE {
+            GRAPH <urn:x-evn-master:clinical_data_dictionary> {
+                ?subject skos:prefLabel ?column_header.
+                ?subject cdd:AttributeType ?attribute_type.
+                ?subject cdd:Datatype ?datatype.
+                ?subject cdd:Description ?description.
+                ?subject cdd:DisplayName ?display_name.
+                ?subject cdd:Priority ?priority.
+            }
+        }
+"""
+
+NUM_FIELDS_IN_CURATED_FILE = 2
+URI_PATTERN_STR = "^C[0-9]{6}$" # e.g. C000000
+URI_PATTERN = re.compile(URI_PATTERN_STR)
+COLUMN_HEADER_PATTERN_STR = "^(\")?[A-Z][A-Z_0-9]{0,99}(\"@en)?$" # e.g. "DELIVERED_DOSE"@en or PLATINUM_OS_MONTHS
+COLUMN_HEADER_PATTERN = re.compile(COLUMN_HEADER_PATTERN_STR)
+MULTIPLE_UNDERSCORES_PATTERN = re.compile("_{2,}")
+errors = []
+warnings = []
+
+def validate_uri(uri, filename):
+    """Adds error to errors array if uri does not match expected pattern"""
+    if not URI_PATTERN.match(uri):
+        errors.append("'%s' does not match expected pattern '%s' in file '%s'" % (uri, URI_PATTERN_STR, filename))
+
+def validate_column_header(str, filename):
+    """Adds error to errors array if str does not match expected pattern"""
+    # validate
+    #   has no whitespace
+    #   must begin with letter
+    #   no lower case letters
+    #   less than or equal to 100 characters
+    #   no characters aside from digits
+    #   upper case letters
+    #   underscore character
+    #   do not have 2 underscores in a row
+    if not COLUMN_HEADER_PATTERN.match(str):
+        errors.append("'%s' does not match expected pattern '%s' in file '%s'" % (str, COLUMN_HEADER_PATTERN_STR, filename))
+    if MULTIPLE_UNDERSCORES_PATTERN.search(str):
+        warnings.append("'%s' contains multiple underscores in file '%s'" % (str, filename))
+
+def read_curated_uris(curated_filename):
+    # validate that file is tab delimited
+    uris = {}
+    with open(curated_filename) as curated_file:
+        for line in curated_file:
+            line = line.strip()
+            if "\t" not in line:
+                errors.append("Line '%s' does not have a '\\t' as a delimiter." % (line))
+            fields = line.split()
+            if len(fields) != NUM_FIELDS_IN_CURATED_FILE:
+                errors.append("Line '%s' has %d field(s), when %d are expectd.  Fields are: %s" % (line, len(fields), NUM_FIELDS_IN_CURATED_FILE, ",".join("'" + field + "'" for field in fields)))
+            else:
+                uris[fields[0]] = fields[1]
+            validate_uri(fields[0], curated_filename)
+            if len(fields) >= 2:
```

Comment:
should we only check this if the length is equal to two? If there's more than two fields it's possible that whatever is in index position 1 is not the column header (plus we are already logging an error about having >2 fields on line 71)